File: gloSuite9023/gloSuite/gloEMRAdmin/gloEMRAdmin/frmPrefix.vb
Imports System.Windows.Forms
Imports Microsoft.Win32
Imports System.Data.SqlClient
Public Class frmPrefix
    Private _ID As Long = 0
    Private _Server As String
    Private _Database As String
    Private _prefix As String

    Public Sub New(ByVal nID As String, ByVal sServer As String, ByVal sDatabase As String, ByVal sPrefix As String)
        InitializeComponent()
        _ID = nID
        _Server = sServer
        _Database = sDatabase
        _prefix = sPrefix

    End Sub
    Public Sub New()
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        ' Add any initialization after the InitializeComponent() call.
    End Sub
    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        Try
            If Trim(txtServer.Text) = "" Then
                MessageBox.Show("Enter server name", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                Exit Sub
            ElseIf Trim(txtDatabase.Text) = "" Then
                MessageBox.Show("Enter database name", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                Exit Sub
            ElseIf txtPrefix.Text.Trim.Length < 3 Then
                MessageBox.Show("Enter 3 digit prefix", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                Exit Sub
            ElseIf checkExists(1) = True Then  ' value 1 to check server and database
                MessageBox.Show("Database and Server name already exists.", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                Exit Sub
            ElseIf checkExists(2) = True Then  ' value 2 to check prefix
                MessageBox.Show("Prefix already in use.", gstrMessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information)
                Exit Sub
            End If
            Save_Prefix()
            Me.Close()
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        End Try
    End Sub
    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub
    Private Sub frmPrefix_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            If _ID > 0 Then
                txtServer.Text = _Server
                txtDatabase.Text = _Database
                txtPrefix.Text = _prefix
            Else
                txtServer.Text = ""
                txtDatabase.Text = ""
                txtPrefix.Text = ""
            End If
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        End Try
    End Sub
    Public Function Save_Prefix() As Boolean
        Try
            Dim objCon As New SqlConnection
            objCon.ConnectionString = gloEMRAdmin.mdlGeneral.GetConnectionString()
            Dim objCmd As New SqlCommand

            objCmd.CommandType = CommandType.StoredProcedure
            objCmd.CommandText = "sp_InUp_Prefix"

            Dim objID As New SqlParameter
            With objID
                .ParameterName = "@nPrefixID"
                .Value = _ID
                .Direction = ParameterDirection.Input
                .SqlDbType = SqlDbType.BigInt
            End With
            objCmd.Parameters.Add(objID)

            Dim objServer As New SqlParameter
            With objServer
                .ParameterName = "@sServer"
                .Value = Trim(txtServer.Text)
                .Direction = ParameterDirection.Input
                .SqlDbType = SqlDbType.VarChar
            End With
            objCmd.Parameters.Add(objServer)

            Dim objDatabase As New SqlParameter
            With objDatabase
                .ParameterName = "@sDatabase"
                .Value = Trim(txtDatabase.Text)
                .Direction = ParameterDirection.Input
                .SqlDbType = SqlDbType.VarChar
            End With
            objCmd.Parameters.Add(objDatabase)

            Dim objPrefix As New SqlParameter
            With objPrefix
                .ParameterName = "@sPrefix"
                .Value = Trim(txtPrefix.Text)
                .Direction = ParameterDirection.Input
                .SqlDbType = SqlDbType.VarChar
            End With
            objCmd.Parameters.Add(objPrefix)

            objCmd.Connection = objCon
            objCon.Open()
            objCmd.ExecuteNonQuery()
            objCon.Close()
            objCmd = Nothing
            objCon = Nothing
            Return True
       
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        End Try
    End Function
    Public Function delete_Prefix(ByVal nprefixId As Long) As Boolean
        Try

       
            Dim objCon As New SqlConnection
            objCon.ConnectionString = gloEMRAdmin.mdlGeneral.GetConnectionString()
            Dim objCmd As New SqlCommand

            objCmd.CommandType = CommandType.StoredProcedure
            objCmd.CommandText = "sp_Delete_Prefix"


            Dim objServer As New SqlParameter
            With objServer
                .ParameterName = "@nPrefixID"
                .Value = nprefixId
                .Direction = ParameterDirection.Input
                .SqlDbType = SqlDbType.BigInt
            End With
            objCmd.Parameters.Add(objServer)

            objCmd.Connection = objCon
            objCon.Open()
            objCmd.ExecuteNonQuery()
            objCon.Close()
            objCmd = Nothing
            objCon = Nothing
            Return True
       
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        End Try
    End Function
    Private Sub txtPrefix_KeyPress(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtPrefix.KeyPress
        ValidateNumeric(txtPrefix.Text, e)
    End Sub
    Public Function ValidateNumeric(ByVal Text As String, ByVal e As KeyPressEventArgs)
        Try
            If Char.IsNumber(e.KeyChar) = False And AscW(e.KeyChar) <> 8 Then
                e.Handled = True
            End If
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        End Try
    End Function
    Public Function checkExists(ByVal FeildName As Integer) As Boolean
        Dim _Query As String
        Dim objcon As New SqlClient.SqlConnection
        objcon.ConnectionString = gloEMRAdmin.mdlGeneral.GetConnectionString()
        Dim cmd As SqlClient.SqlCommand
        Dim _blnResult As Boolean = False
        Dim _strResult As Integer = 0

        Try
            If FeildName = 1 Then
                _Query = "SELECT COUNT(*) FROM Prefix WHERE sServer = '" & Trim(txtServer.Text) & "' AND sDatabase= '" & Trim(txtDatabase.Text) & "' "
            Else
                _Query = "SELECT COUNT(*) FROM Prefix WHERE sPrefix = '" & Trim(txtPrefix.Text) & "'"
            End If

            cmd = New SqlClient.SqlCommand(_Query, objcon)

            objcon.Open()
            _strResult = cmd.ExecuteScalar
            objcon.Close()
            If _strResult > 0 Then
                _blnResult = True
            End If

            Return _blnResult
        Catch ex As Exception
            gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), False)
        Finally
            objcon = Nothing
        End Try
    End Function
End Class

File: gloSuite9023/gloSuite/gloCommon/gloSettings/frmPrefix.cs
using System;

using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient ;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace gloSettings 
{
    public partial class frmPrefix : Form
    {   
        private long _ID = 0;
       // private string prev_prefix = null;
        private string _Server=null;
        private string _Database=null;
        private String _prefix=null;
        private string _databaseconnectionstring = "";

        enum RecordExists
        {
            ServerAndDatabase,
            Prefix
        }

        public frmPrefix(string databaseconnectionstring)
        {
            _databaseconnectionstring = databaseconnectionstring;
            InitializeComponent();
        }
        public frmPrefix(long nID, string sServer, string sDatabase, string sPrefix, string databaseconnectionstring)
        {

            _databaseconnectionstring = databaseconnectionstring;
            InitializeComponent();
            _ID = nID;
    //         prev_prefix= _prefix ;
            _Server = sServer;
            _Database = sDatabase;
            _prefix = sPrefix;

        }
        private void frmPrefix_Load(object sender, EventArgs e)
        {
            if (_ID > 0)
            {
                txtServer.Text   = _Server;
                txtDatabase.Text = _Database;
                txtPrefix.Text   = _prefix;
            }
            else
            {
                txtServer.Text = "";
                txtDatabase.Text = "";
                txtPrefix.Text = "";
            }

        }

        public bool Save_Prefix()
        {
            SqlCommand objCmd = null;
            try
            {

                SqlConnection objCon = new SqlConnection();
                objCon.ConnectionString = _databaseconnectionstring;
                objCmd = new SqlCommand();

                objCmd.CommandType = CommandType.StoredProcedure;
                objCmd.CommandText = "gsp_InUp_Prefix";

                SqlParameter objID = new SqlParameter();
                {
                    objID.ParameterName = "@nPrefixID";
                    objID.Value = _ID;
                    objID.Direction = ParameterDirection.Input;
                    objID.SqlDbType = SqlDbType.BigInt;
                }
                objCmd.Parameters.Add(objID);
                objID = null;

                SqlParameter objServer = new SqlParameter();
                {
                    objServer.ParameterName = "@sServer";
                    objServer.Value = (txtServer.Text.Trim());
                    objServer.Direction = ParameterDirection.Input;
                    objServer.SqlDbType = SqlDbType.VarChar;
                }
                objCmd.Parameters.Add(objServer);
                objServer = null;

                SqlParameter objDatabase = new SqlParameter();
                {

                    objDatabase.ParameterName = "@sDatabase";
                    objDatabase.Value = (txtDatabase.Text.Trim());
                    objDatabase.Direction = ParameterDirection.Input;
                    objDatabase.SqlDbType = SqlDbType.VarChar;
                }
                objCmd.Parameters.Add(objDatabase);
                objDatabase = null;


                SqlParameter objPrefix = new SqlParameter();
                {
                    objPrefix.ParameterName = "@sPrefix";
                    objPrefix.Value = txtPrefix.Text.Trim();
                    objPrefix.Direction = ParameterDirection.Input;
                    objPrefix.SqlDbType = SqlDbType.VarChar;
                }
                objCmd.Parameters.Add(objPrefix);
                objPrefix = null;


                objCmd.Connection = objCon;
                objCon.Open();
                objCmd.ExecuteNonQuery();
                objCon.Close();


                if (_ID > 0)
                {
                    string strDescription;
                    strDescription = _Server + " " + _Database + " " + _prefix + " " + "modified to" + " " + txtServer.Text + " " + txtDatabase.Text + " " + txtPrefix.Text;
                    gloAuditTrail.gloAuditTrail.CreateAuditLog(gloAuditTrail.ActivityModule.Setting, gloAuditTrail.ActivityCategory.Settings, gloAuditTrail.ActivityType.Modify, strDescription, 0, 0, 0, gloAuditTrail.ActivityOutCome.Success);

                }
                else
                {
                    gloAuditTrail.gloAuditTrail.CreateAuditLog(gloAuditTrail.ActivityModule.Setting, gloAuditTrail.ActivityCategory.Settings, gloAuditTrail.ActivityType.Add, "New prefix Added", 0, 0, 0, gloAuditTrail.ActivityOutCome.Success);

                }
                //objCmd = null;
                objCon = null;
                //return true;

            }
            catch (Exception ex)
            {
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);

            }
            finally
            {
                if (objCmd != null)
                {
                    objCmd.Parameters.Clear();  
                    objCmd.Dispose();
                    objCmd = null;

                }
            }
            return true;
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            RecordExists checkPrefix = RecordExists.Prefix;
            RecordExists CheckServer = RecordExists.ServerAndDatabase;
            try
            {

                if (string.IsNullOrEmpty((txtServer.Text.Trim())))
                {

                    MessageBox.Show("Enter server name", gloAuditTrail.gloAuditTrail.MessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                    txtServer.Focus();  
                    return;
                }
                else if (string.IsNullOrEmpty((txtDatabase.Text.Trim())))
                {
                    MessageBox.Show("Enter database name", gloAuditTrail.gloAuditTrail.MessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                    txtDatabase.Focus();  
                    return;
                }
                else if (txtPrefix.Text.Trim().Length < 3)
                {
                    MessageBox.Show("Enter 3 character site prefix", gloAuditTrail.gloAuditTrail.MessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                    return;

                }

                else if (txtPrefix.Text != _prefix)
                {
                    if (checkExists(checkPrefix.ToString()) == true)
                    {
                        MessageBox.Show("Prefix already in use.", gloAuditTrail.gloAuditTrail.MessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                        txtPrefix.Focus();  
                        return;
                    }

                }
                if ((txtDatabase.Text != _Database) || (txtServer.Text != _Server))
                {
                    if (checkExists(CheckServer.ToString()) == true)
                    {
                        MessageBox.Show("Database and Server name already exists.", gloAuditTrail.gloAuditTrail.MessageBoxCaption, MessageBoxButtons.OK, MessageBoxIcon.Information);
                        txtDatabase.Focus();
                        return;

                    }
                    Save_Prefix();
                }
                else
                {
                    Save_Prefix();
                }
               
                this.Close();
            }
            catch (Exception ex)
            {
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
            }
            finally
            {
               
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        public bool checkExists(string FeildName)
        {
            string _Query = null;
            System.Data.SqlClient.SqlConnection objcon = new System.Data.SqlClient.SqlConnection();
            objcon.ConnectionString = _databaseconnectionstring;
            System.Data.SqlClient.SqlCommand cmd = default(System.Data.SqlClient.SqlCommand);
            bool _blnResult = false;
            int _strResult = 0;

            try
            {
                if (FeildName == RecordExists.ServerAndDatabase.ToString())
                {
                    _Query = "SELECT COUNT(*) FROM Prefix WHERE sServer = '" + (txtServer.Text.Trim()) + "' AND sDatabase= '" + txtDatabase.Text.Trim() + "' ";
                }
                else if (FeildName == RecordExists.Prefix.ToString())
                {
                    _Query = "SELECT COUNT(*) FROM Prefix WHERE sPrefix = '" + (txtPrefix.Text.Trim()) + "'";
                }

                cmd = new System.Data.SqlClient.SqlCommand(_Query, objcon);

                objcon.Open();
                _strResult = Convert.ToInt32(cmd.ExecuteScalar());
                objcon.Close();
                if (_strResult > 0)
                {
                    _blnResult = true;
                }

                //return _blnResult;
            }
            catch (Exception ex)
            {
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
                _blnResult = false;
            }
            finally
            {
                if (objcon != null)
                {
                    objcon.Dispose();
                    objcon = null;
                }
                if (cmd != null)
                {
                    cmd.Parameters.Clear();  
                    cmd.Dispose();
                    cmd = null;
                }
                _Query = null;
            }
            return _blnResult;
        }
        public bool delete_Prefix(long nprefixId)
        {
            SqlParameter objServer =null;
            SqlCommand objCmd =null;
            try
            {

                SqlConnection objCon = new SqlConnection();
                objCon.ConnectionString = _databaseconnectionstring;
                objCmd = new SqlCommand();

                objCmd.CommandType = CommandType.StoredProcedure;
                objCmd.CommandText = "gsp_Delete_Prefix";
                
                objServer = new SqlParameter();
                {
                    objServer.ParameterName = "@nPrefixID";
                    objServer.Value = nprefixId;
                    objServer.Direction = ParameterDirection.Input;
                    objServer.SqlDbType = SqlDbType.BigInt;
                }
                objCmd.Parameters.Add(objServer);

                objCmd.Connection = objCon;
                objCon.Open();
                objCmd.ExecuteNonQuery();
                objCon.Close();
                gloAuditTrail.gloAuditTrail.CreateAuditLog(gloAuditTrail.ActivityModule.Setting, gloAuditTrail.ActivityCategory.Settings, gloAuditTrail.ActivityType.Delete, " Prefix deleted  ", 0, 0, 0, gloAuditTrail.ActivityOutCome.Success);
                //objCmd = null;
                objServer = null;
                objCon = null;
                return true;

            }
            catch (Exception ex)
            {
                gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
            }
            finally
            {
                if (objCmd != null)
                {
                    objCmd.Parameters.Clear();
                    objCmd.Dispose();
                    objCmd = null;

                }
            }
            return true;
        }


            private void txtPrefix_KeyPress(object sender, KeyPressEventArgs e)
            {
                try
                {
                    if (char.IsNumber(e.KeyChar) == false && (e.KeyChar) != 8 && (char.IsLetter(e.KeyChar) == false))
                    {
                        e.Handled = true;
                    }
                }
                catch (Exception ex)
                {
                    gloAuditTrail.gloAuditTrail.ExceptionLog(ex.ToString(), false);
                }
            }
}
    }





